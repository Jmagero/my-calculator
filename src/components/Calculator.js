/* eslint-disable */
import React, { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({ total: '0' });
  const handleClick = (event) => {
    const text = event.target.value;
    const objstate = state;
    const newState = calculate(objstate, text);
    setState(newState);
    const calculatorDisplay = document.querySelector('.calculator-display');
    if (newState.total === null && newState.next === null && newState.operation === null) {
      calculatorDisplay.innerHTML = '0';
    } else if (newState.next !== null) {
      calculatorDisplay.textContent = newState.next;
    } else if (newState.next === null && newState.total !== null) {
      calculatorDisplay.textContent = newState.total;
    }
  };
  return (
    <div className="wrapper"
    style={{
      display: 'flex',
      justifyContent: "space-between"
    }}>
       <h3>Let's do some Math!</h3>
       <div className="button-wrapper">
        <div className="button-row calculator-display">0</div>
        <div className="button-row">
          <button type="button" value="AC" onClick={handleClick}>AC</button>
          <button type="button" value="+/-" onClick={handleClick}>+/-</button>
          <button type="button" value="%" onClick={handleClick}>%</button>
          <button className="button-right" type="button" value="รท" onClick={handleClick}>รท</button>
        </div>
        <div className="button-row">
          <button type="button" value="7" onClick={handleClick}>7</button>
          <button type="button" value="8" onClick={handleClick}>8</button>
          <button type="button" value="9" onClick={handleClick}>9</button>
          <button className="button-right" type="button" value="x" onClick={handleClick}>*</button>
        </div>
        <div className="button-row">
          <button type="button" value="4" onClick={handleClick}>4</button>
          <button type="button" value="5" onClick={handleClick}>5</button>
          <button type="button" value="6" onClick={handleClick}>6</button>
          <button type="button" value="-" onClick={handleClick} className="button-right">-</button>
        </div>
        <div className="button-row">
          <button type="button" value="1" onClick={handleClick}>1</button>
          <button type="button" value="2" onClick={handleClick}>2</button>
          <button type="button" value="3" onClick={handleClick}>3</button>
          <button className="button-right" type="button" value="+" onClick={handleClick}>+</button>
        </div>
        <div className="button-row">
          <button className="zero" type="button" value="0" onClick={handleClick}>0</button>
          <button type="button" value="-" onClick={handleClick}>-</button>
          <button className="button-right" type="button" value="=" onClick={handleClick}>=</button>
        </div>
       </div>
    </div>
  );
};

export default Calculator;
