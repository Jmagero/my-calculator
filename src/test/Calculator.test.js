import React from 'react';
import renderer from 'react-test-renderer';
import { screen, render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Calculator from '../components/Calculator';

it('renders correctly', () => {
  const tree = renderer.create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders "AC" button in Calculator component', () => {
  render(<Calculator />);
  expect(screen.getByText('AC')).toBeInTheDocument();
});

it('renders "=" button in Calculator component', () => {
  render(<Calculator />);
  expect(screen.getByText('=')).toBeInTheDocument();
});
