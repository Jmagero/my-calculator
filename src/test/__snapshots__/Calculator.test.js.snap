// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "display": "flex",
      "justifyContent": "space-between",
    }
  }
>
  <h3>
    Let's do some Math!
  </h3>
  <div
    className="button-wrapper"
  >
    <div
      className="button-row calculator-display"
    >
      0
    </div>
    <div
      className="button-row"
    >
      <button
        onClick={[Function]}
        type="button"
        value="AC"
      >
        AC
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="+/-"
      >
        +/-
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="%"
      >
        %
      </button>
      <button
        className="button-right"
        onClick={[Function]}
        type="button"
        value="รท"
      >
        รท
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        onClick={[Function]}
        type="button"
        value="7"
      >
        7
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="8"
      >
        8
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="9"
      >
        9
      </button>
      <button
        className="button-right"
        onClick={[Function]}
        type="button"
        value="x"
      >
        *
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        onClick={[Function]}
        type="button"
        value="4"
      >
        4
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="5"
      >
        5
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="6"
      >
        6
      </button>
      <button
        className="button-right"
        onClick={[Function]}
        type="button"
        value="-"
      >
        -
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        onClick={[Function]}
        type="button"
        value="1"
      >
        1
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="2"
      >
        2
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="3"
      >
        3
      </button>
      <button
        className="button-right"
        onClick={[Function]}
        type="button"
        value="+"
      >
        +
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="zero"
        onClick={[Function]}
        type="button"
        value="0"
      >
        0
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="-"
      >
        -
      </button>
      <button
        className="button-right"
        onClick={[Function]}
        type="button"
        value="="
      >
        =
      </button>
    </div>
  </div>
</div>
`;
